client/components/tinymce/plugins/wpcom/test/utils.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import { assert } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import { removeEmptySpacesInParagraphs } from '../wpcom-utils';
+
+describe( 'remove empty spaces in paragraphs', () => {
+	it( 'should not modify paragraphs with content', () => {
+		const content = '<p>Chicken &amp; Ribs</p>';
+
+		assert.equal( content, removeEmptySpacesInParagraphs( content ) );
+	} );
+
+	it( 'should remove &nbsp; from empty paragraphs', () => {
+		const content = '<p>&nbsp;</p>';
```

Comment:
Yeah, don't think the behavior should change for broken HTML. `<p><p>&nbsp;</p></p>` will be replaced with `<p><p><br></p></p>` which in the editor DOM will become `<p></p><p><br></p><p></p>` (the closing `</p>` is optional, so the first paragraph is valid, the second is valid too, and the browser will insert the third `<p>` as "fix").